addons/l10n_mn/migrations/1.1/pre-migrate.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details
+
+from odoo import api, SUPERUSER_ID
+
+
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+
+    # Account tags: all account tags to be deleted should be unset from accounts
+    cr.execute(r'''
+        DELETE FROM account_account_account_tag aaat
+        USING account_account_tag
+        JOIN ir_model_data
+            ON ir_model_data.module = 'l10n_mn'
+            AND ir_model_data.model = 'account.account.tag'
+            AND ir_model_data.res_id = account_account_tag.id
+        WHERE aaat.account_account_tag_id = account_account_tag.id
+            AND (
+                ir_model_data.name LIKE 'tax\_report\_tag%'
+                OR ir_model_data.name LIKE 'account\_tag\_operating\_%'
+                OR ir_model_data.name LIKE 'account\_tag\_investing\_%'
+                OR ir_model_data.name LIKE 'account\_tag\_financing\_%'
+                OR ir_model_data.name IN ('account_tag_exchange', 'vat_report_tag58', 'vat_report_tag59')
+            )
+    ''')
+
+    # Delete old account tags
+    cr.execute(r'''
+        DELETE FROM account_account_tag
+        USING ir_model_data
+        WHERE
+            account_account_tag.id = ir_model_data.res_id
+            AND ir_model_data.module = 'l10n_mn'
+            AND ir_model_data.model = 'account.account.tag'
+            AND (
+                ir_model_data.name LIKE 'tax\_report\_tag%'
+                OR ir_model_data.name LIKE 'account\_tag\_operating\_%'
+                OR ir_model_data.name LIKE 'account\_tag\_investing\_%'
+                OR ir_model_data.name LIKE 'account\_tag\_financing\_%'
+                OR ir_model_data.name IN ('account_tag_exchange', 'vat_report_tag58', 'vat_report_tag59')
+            )
+    ''')
```

Comment:
All these tags have `applicability=accounts` so they will never be used on AMLs, but only on accounts. The Many2Many table between accounts and account tags has an ONDELETE SET NULL so this will not raise.

I remove the old VAT report explicitly here because the codes I used on the report lines of the new VAT report are similar (`VAT_LINExx`). Since the new report is in module `l10n_mn` (and therefore has a different XMLid than the old report), the XML that defines the new VAT report will create new report lines, but if they have the same codes as the existing report lines in the old report, this will trigger the SQL constraint that all report line codes must be different.

Since the old report has been removed from the data in `l10n_mn_reports`, it would be garbage-collected anyway at step 4 of the module-loading process (unless it was manually set to noupdate), so deleting is here is merely hastening that. 

I do note the point about people potentially depending on the old report, and I agree that is a good argument for not doing this in stable.